media-api/app/lib/elasticsearch/SearchFilters.scala

```diff
@@ -90,4 +79,9 @@ trait SearchFilters extends ImageFields {
     "pool"
   )
 
+  def filterOrFilter(filter: Option[FilterBuilder], orFilter: Option[FilterBuilder]): Option[FilterBuilder] = (filter, orFilter) match {
```

Comment:
You could have some implicits potentially:
``` Scala
implicit class FilterBuilderHelpers(filter: Option[FilterBuilder]) {
    def ||(orFilter: Option[FilterBuilder]): Option[FilterBuilder] = (filter, orFilter) match {
      case (Some(someFilter), Some(orSomeFilter)) => Some(filters.or(someFilter, orSomeFilter))
      case (filterOpt,    orFilterOpt)    => filterOpt orElse orFilterOpt
    }
  }
```

and then the implementation looks like:
```freeMetadataFilter || freeUsageRightsFilter```

But that's still got the weird pattern matching code.

I need to look more into [elastic4s](https://github.com/sksamuel/elastic4s)...